From 52cf14a29dbaf9e2deff36767443d2a9d404e117 Mon Sep 17 00:00:00 2001
From: Roman Lebedev <lebedev.ri@gmail.com>
Date: Tue, 20 Aug 2024 06:51:30 +0300
Subject: [PATCH 1/2] Fix `chuffed-config.cmake.in`

While it (correctly!) deals with relative (instead of absolute) paths,
there still is a problem: it assumes that `${CMAKE_CURRENT_LIST_DIR}`
is `<???>/usr/lib/cmake/chuffed/`, but in e.g. debian it's actually
`<???>/usr/lib/x86_64-linux-gnu/cmake/chuffed/`,
which requires 4, not 3, `cd ..`.

Fix that by computing the relative path at CMake time,
oh, and avoid using `CMAKE_INSTALL_INCLUDEDIR` et al
that may be different when the config is consumed...

Forwarded: https://github.com/chuffed/chuffed/pull/198
---
 CMakeLists.txt          | 3 +++
 chuffed-config.cmake.in | 4 ++--
 2 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/chuffed/CMakeLists.txt a/chuffed/CMakeLists.txt
index 8db276d82..0b032569c 100644
--- a/chuffed/CMakeLists.txt
+++ a/chuffed/CMakeLists.txt
@@ -392,6 +392,9 @@ install(FILES ${CMAKE_BINARY_DIR}/chuffed.msc DESTINATION ${CMAKE_INSTALL_DATARO
 # ------------- CMake export configuration -------------
 include(CMakePackageConfigHelpers)
 
+file(RELATIVE_PATH REL_INCLUDEDIR ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}/cmake/chuffed/ ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_INCLUDEDIR})
+file(RELATIVE_PATH REL_DATAROOTDIR ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}/cmake/chuffed/ ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_DATAROOTDIR})
+
 export(EXPORT chuffed-targets FILE ${CMAKE_CURRENT_BINARY_DIR}/cmake/chuffed-targets.cmake)
 
 configure_package_config_file(
diff --git a/chuffed/chuffed-config.cmake.in a/chuffed/chuffed-config.cmake.in
index 7fd432f1d..e1762bca9 100644
--- a/chuffed/chuffed-config.cmake.in
+++ a/chuffed/chuffed-config.cmake.in
@@ -5,6 +5,6 @@ include("${CMAKE_CURRENT_LIST_DIR}/chuffed-targets.cmake")
 check_required_components(chuffed chuffed_fzn)
 
 set(CHUFFED_LIBRARIES chuffed_fzn chuffed)
-set(CHUFFED_INCLUDE_DIRS ${CMAKE_CURRENT_LIST_DIR}/../../../${CMAKE_INSTALL_INCLUDEDIR})
-set(CHUFFED_SHARE_DIR ${CMAKE_CURRENT_LIST_DIR}/../../../${CMAKE_INSTALL_DATAROOTDIR})
+set(CHUFFED_INCLUDE_DIRS ${CMAKE_CURRENT_LIST_DIR}/@REL_INCLUDEDIR@)
+set(CHUFFED_SHARE_DIR ${CMAKE_CURRENT_LIST_DIR}/@REL_DATAROOTDIR@)
 set(CHUFFED_FOUND TRUE)
-- 
2.45.2

